src/praisonai-agents/praisonaiagents/process/process.py

```
@@ -883,10 +914,15 @@ class ManagerInstructions(BaseModel):
             if selected_task_id not in self.tasks:
-                error_msg = f"Manager selected invalid task id {selected_task_id}"
-                display_error(error_msg)
-                logging.error(error_msg)
-                break
+                # Re-prompt the manager with valid task IDs instead of terminating
+                valid_task_ids = list(self.tasks.keys())
+                error_msg = f"Manager selected invalid task_id={selected_task_id}. Valid task IDs are: {valid_task_ids}"
+                logging.warning(error_msg)
+                manager_prompt += (
+                    f"\n\n[ERROR] Your previous selection of task_id={selected_task_id} was invalid. "
+                    f"Valid task IDs are: {valid_task_ids}. Please select again from the valid options."
```

Ruff (0.15.11)

[error] Possible SQL injection vector through string-based query construction

(S608)